### DISCUSSION TITLE
Disable password hint isn't working for me

### DISCUSSION THREAD

**Discussion by ShlomiD83** [dani-garcia/vaultwarden#5179] [Q&A] [answered]
Hi guys, 
I'm trying to disable the password hint feature but the feature won't disable.
My config.json shows the option as disabled
`password_hints_allowed": false,
  "show_password_hint": false,`

If I check the admin portal I see the option as disabled as well but when I press the Get master password hint it sends the hint email.

How to disable it completely?

**Comment by BlackDex** [comment]
I'm guessing that you think it isn't working because you still see the options on the web-vault interface etc...
That is because we currently do not support dynamic CSS changes (yet) which could hide those elements from view.

But rest assured, the feature will not work and produces error's or empty responses where needed.

**Reply by ShlomiD83** [reply]
Are you sure? Because I'm getting an email with my password hint.....

**Reply by BlackDex** [reply, marked as answer]
A wrong if check was done. I fixed that just now and a PR is awaiting for merge now.

**Reply by ShlomiD83** [reply]
Now it throws an error, thanks.

**Comment by jjlin** [comment]
I don't think this change is correct actually. If you read what these settings are documented to do:

* `show_password_hint` -- "Controls whether a password hint should be shown directly in the web page if SMTP service is not configured."
* `password_hints_allowed` -- "Controls whether users can _*set*_ password hints."

This means that `password_hints_allowed` will prevent a user from setting a new password hint, but if a password hint was already set prior to `password_hints_allowed` being disabled, it will still be returned to the user. From a user-centric view, you could argue that it would be inappropriate to unilaterally remove their access to a hint they may have been depending on. From an admin-centric view, you could argue that the admin doesn't care how a user feels about this.

**Reply by stefan0xC** [reply]
I concur that the change is not correct because the description specifies that it would only affect systems where SMTP is not configured:
https://github.com/dani-garcia/vaultwarden/blob/e927b8aa5ec8f0352bfb2ff95a996a89389959ff/src/config.rs#L502-L505

At the moment the change in #5180 https://github.com/dani-garcia/vaultwarden/blob/e927b8aa5ec8f0352bfb2ff95a996a89389959ff/src/api/core/accounts.rs#L845 disables the password hint for anyone that has not enabled mails and turned it into a switch for when you have mail enabled.

I think we have a few options: 

1.  `show_password_hint` should determine this for both options (so whether you have mail enabled or not would not matter) in which case we should update the description accordingly, 
2.  we return to the previous state, where mail enabled systems would sent already configured password hints to users (even if admin later decides to forbid setting password hints)
3. or we change it back and let `password_hints_allowed` decide instead whether password hints will be sent even if it's already configured instead of `show_password_hint`).

My preference would probably be the third option (if I don't allow password hints I don't want this to work), because the first seems a bit silly as we do have another option for that, and the `show_password_hint` option would only be relevant for systems that don't have mail enabled (which you have to activate).
